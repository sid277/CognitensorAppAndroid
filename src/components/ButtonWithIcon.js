import React from 'react';
import { View, TouchableOpacity, StyleSheet, Text } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { theme } from '../theme';

const ButtonWithIcon = ({ title, iconName, onPress }) => {
  return (
    <TouchableOpacity style={styles.buttonContainer} onPress={() => onPress()}>
      <View style={styles.buttonViewContainer}>
        <MaterialIcons name={iconName} size={24} color="#C4C4C4" />
        <Text style={styles.buttonText}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    alignSelf: 'stretch',
  },
  buttonViewContainer: {
    flexDirection: 'row',
    backgroundColor: '#FBFBFB',
    borderBottomWidth: 1.5,
    borderColor: 'rgba(221, 221, 221, 0.24)',
    alignSelf: 'stretch',
    alignItems: 'center',
    padding: theme.spacing.base,
    ...theme.typography.body,
  },
  buttonText: {
    marginLeft: theme.spacing.small,
    color: '#6C6C6C',
  },
});

export default ButtonWithIcon;
