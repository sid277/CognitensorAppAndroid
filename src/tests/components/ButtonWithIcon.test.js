import React from 'react';
import { shallow } from 'enzyme';
import ButtonWithIcon from '../../components/ButtonWithIcon';
import renderer from 'react-test-renderer';


test('should test AppHeader component', () => {
const wrapper = shallow(<ButtonWithIcon />);
expect(wrapper).toMatchSnapshot();
});