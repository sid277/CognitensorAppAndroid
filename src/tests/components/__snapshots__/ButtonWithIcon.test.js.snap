// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test AppHeader component 1`] = `ShallowWrapper {}`;
